src/class/adapter/memory.adapter.ts

```diff
@@ -1,37 +1,86 @@
+// Import Third-party Dependencies
+import { Result, Err, Ok } from "@openally/result";
+import EphemeralMap, { tSv } from "@openally/ephemeral-map";
+
 // Import Internal Dependencies
-import type { DatabaseConnection, GetConnectionPerfResponse } from "../../types";
+import type {
+  DatabaseConnection,
+  KeyType
+} from "../../types";
+import { SetValueError } from "../error/memory.adapter.error";
+
+export interface InMemSetValueOptions {
+  key: string;
+  value: unknown;
+  expiresIn?: number;
+}
+
+export interface InMemIsKeyExpiredOptions {
+  value: Record<string, unknown>;
+  banTimeInSecond: number;
+}
 
 export class MemoryAdapter implements DatabaseConnection {
-  async initialize(): Promise<void> {
-    // Simple connection logic
-  }
+  #values: EphemeralMap<string, unknown> = new EphemeralMap();
 
-  async close(forceExit?: boolean): Promise<void> {
-    // Simple close connection logic
+  flushall() {
+    this.#values = new EphemeralMap();
   }
 
-  async isAlive(): Promise<boolean> {
-    return true;
-  }
+  setValue(options: InMemSetValueOptions): Result<KeyType, SetValueError> {
+    const { key, value, expiresIn } = options;
 
-  async getPerformance(): Promise<GetConnectionPerfResponse> {
-    return { isAlive: true, perf: 100 };
-  }
+    const valueExist = this.#values.has(key);
+
+    if (valueExist) {
+      return Err(new SetValueError(key));
+    }
+
+    this.#values.set(tSv({ ttl: expiresIn })(key), value);
 
-  async setValue(): Promise<any> {
-    // Set value logic
+    return Ok(key);
   }
 
-  async deleteValue(): Promise<any> {
-    // Delete value logic
+  deleteValue(key: string) {
+    const isDelete = this.#values.delete(key);
+
+    return isDelete ? 1 : 0;
   }
```

Comment:
Pas mieux de retourner un boolean ? `return Boolean(this.#values.delete(key))`

Reply:
Certainement car la méthode original de redis return soi 1 ou 0 (c'est un adapter donc ça doit être équivalent)